ethcore/src/miner/miner.rs

```diff
@@ -482,6 +489,7 @@ impl Miner {
 				// Save proposal for later seal submission and broadcast it.
 				Seal::Proposal(seal) => {
 					trace!(target: "miner", "Received a Proposal seal.");
+					*self.next_mandatory_reseal.write() = Instant::now() + self.options.reseal_max_period;
```

Comment:
Indeed, it would start sealing on transactions after 2 min. Either can make this period very large or make it specific for internal sealing, which I think makes more sense.